Raven.Database/Bundles/Replication/Controllers/ReplicationController.cs

```diff
@@ -562,7 +578,14 @@ private void SaveReplicationSource(string src, string lastEtag, int batchSize, s
                                  new RavenJObject(), null);
                 });
             }
-
+            //We don't want to delay large batches of attachments otherwise we will endup with alot of data that needs to get replicated.
+            if (Request.Content.Headers.ContentLength > Database.Configuration.Replication.ReplicationPropagationDelaySizeInBytes)
+            {
```

Comment:
I have removed it, but from what i know batches is according to # documents and not their sizes...
If the documents are from replication their size of disk is unknown to the prefetcher...